### PR TITLE
added Polly40 keyboard

### PR DESCRIPTION
**Pull request by PollyV1**
## Description
added Polly 40 keyboard

## Types of Changes

- [x] Core
- [x] Bugfix
- [ ] New feature
- [x] Enhancement/optimization
- [x] Keyboard (addition or update)
- [x] Keymap/layout/userspace (addition or update)
- [x] Documentation

## Issues Fixed or Closed by This PR

* 

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [x] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [x] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: keyboards/handwired/polly40/config.h [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+
+#pragma once
+#define DYNAMIC_KEYMAP_LAYER_COUNT 4
+#include "config_common.h"
+
+/* USB Device descriptor parameter */
+
+/* key matrix size */
+#define MATRIX_ROWS 4
+#define MATRIX_COLS 12
```
Comment (fauxpark) [inline keyboards/handwired/polly40/config.h:25]:
```suggestion
```

File: keyboards/handwired/polly40/config.h [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+ * ROWS: AVR pins used for rows, top to bottom
+ * DIODE_DIRECTION: COL2ROW = COL = Anode (+), ROW = Cathode (-, marked on diode)
+ *                  ROW2COL = ROW = Anode (+), COL = Cathode (-, marked on diode)
+ *
+ */
+#define MATRIX_ROW_PINS { F0, F1, F5, B4 }
+#define MATRIX_COL_PINS { F4, D7, B5, B6, C6, C7, D4, D6, D5, D0, D1, D2 }
+
+#define DIODE_DIRECTION COL2ROW
+#define DEBOUNCE 5
```
Comment (fauxpark) [inline keyboards/handwired/polly40/config.h:41]:
```suggestion
```

File: keyboards/handwired/polly40/rules.mk [hunk 3]
```diff
@@ -0,0 +1,25 @@
+# MCU name
+MCU = atmega32u4
+
+# Bootloader selection
+BOOTLOADER = halfkay
+
```
Comment (fauxpark) [inline keyboards/handwired/polly40/rules.mk:6]:
```suggestion
```

File: keyboards/handwired/polly40/info.json [hunk 4]
```diff
@@ -0,0 +1,58 @@
+{
+    "keyboard_name": "polly40 v2",
+    "manufacturer": "paul",
+    "url": "https://github.com/PollyV1",
+    "maintainer": "Paul Enrico N. Viola",
+    "usb": {
+        "vid": "0x6532",
+        "pid": "0x0069",
+        "device_version": "0.0.1"
+    },
```
Comment (fauxpark) [inline keyboards/handwired/polly40/info.json:10]:
```suggestion
    },
    "processor": "atmega32u4",
    "bootloader": "caterina",
```

File: keyboards/handwired/polly40/polly40.h [hunk 5]
```diff
@@ -0,0 +1,72 @@
...
+    K30, K31, K32, K36, K3A, K3B  \
+) { \
+    { K00, K01, K02, K03, K04, K05, K06, K07, K08, K09, K0A, K0B }, \
+    { K10, K11, K12, K13, K14, K15, K16, K17, K18, K19, XXX, K1B }, \
+    { K20, XXX, K22, K23, K24, K25, K26, K27, K28, K29, K2A, K2B }, \
+    { K30, K31, K32, XXX, XXX, XXX, K36, XXX, XXX, XXX, K3A, K3B }, \
+}
+
+// generated by KBFirmware JSON to QMK Parser
+// https://noroadsleft.github.io/kbf_qmk_converter/
```
Comment (fauxpark) [inline keyboards/handwired/polly40/polly40.h:72]:
```suggestion
```

File: keyboards/handwired/polly40/rules.mk [hunk 6]
```diff
@@ -0,0 +1,25 @@
...
+EXTRAKEY_ENABLE = yes       # Audio control and System control
+CONSOLE_ENABLE = yes        # Console for debug
+COMMAND_ENABLE = yes        # Commands for debug and configuration
+# Do not enable SLEEP_LED_ENABLE. it uses the same timer as BACKLIGHT_ENABLE
+SLEEP_LED_ENABLE = no       # Breathing sleep LED during USB suspend
+# if this doesn't work, see here: https://github.com/tmk/tmk_keyboard/wiki/FAQ#nkro-doesnt-work
+NKRO_ENABLE = yes            # USB Nkey Rollover
+BACKLIGHT_ENABLE = no       # Enable keyboard backlight functionality
+RGBLIGHT_ENABLE = no        # Enable keyboard RGB underglow
+BLUETOOTH_ENABLE = no       # Enable Bluetooth
```
Comment (fauxpark) [inline keyboards/handwired/polly40/rules.mk:21]:
```suggestion
NKRO_ENABLE = yes           # Enable N-Key Rollover
BACKLIGHT_ENABLE = no       # Enable keyboard backlight functionality
RGBLIGHT_ENABLE = no        # Enable keyboard RGB underglow
```

**Review by fauxpark — CHANGES_REQUESTED** [review]

File: keyboards/handwired/polly40/readme.md [hunk 7]
```diff
@@ -0,0 +1,17 @@
+# Polly40
+
+A handwired keyboard on a Pro micro Atmega32u4 controller
+
+![ Polly ](https://i.imgur.com/QJCQcFA.jpg)
```
Comment (fauxpark) [inline keyboards/handwired/polly40/readme.md:5]:
```suggestion
![ Polly ](https://i.imgur.com/QJCQcFAh.jpg)
```

**Review by fauxpark — CHANGES_REQUESTED** [review]

**Comment by PollyV1** [comment]
is it possible to add via keymaps or no?

File: keyboards/handwired/polly40/config.h [hunk 8]
```diff
@@ -0,0 +1,41 @@
...
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#pragma once
+#define DYNAMIC_KEYMAP_LAYER_COUNT 4
```
Comment (fauxpark) [inline keyboards/handwired/polly40/config.h:18]:
```suggestion
```

File: keyboards/handwired/polly40/config.h [hunk 9]
```diff
@@ -0,0 +1,41 @@
+/* Copyright 2021 REPLACE_WITH_YOUR_NAME
```
Comment (fauxpark) [inline keyboards/handwired/polly40/config.h:1]: Please fill these out.

**Review by fauxpark — CHANGES_REQUESTED** [review]

File: keyboards/handwired/polly40/config.h [hunk 10]
```diff
@@ -0,0 +1,40 @@
...
+ * COLS: AVR pins used for columns, left to right
+ * ROWS: AVR pins used for rows, top to bottom
+ * DIODE_DIRECTION: COL2ROW = COL = Anode (+), ROW = Cathode (-, marked on diode)
+ *                  ROW2COL = ROW = Anode (+), COL = Cathode (-, marked on diode)
+ *
+ */
+#define MATRIX_ROW_PINS { F0, F1, F5, B4 }
+#define MATRIX_COL_PINS { F4, D7, B5, B6, C6, C7, D4, D6, D5, D0, D1, D2 }
+
+#define DIODE_DIRECTION COL2ROW
```
Comment (waffle87) [inline keyboards/handwired/polly40/config.h:32]:
```suggestion
#define MATRIX_ROW_PINS { F0, F1, F5, B4 }
#define MATRIX_COL_PINS { F4, D7, B5, B6, C6, C7, D4, D6, D5, D0, D1, D2 }
```

File: keyboards/handwired/polly40/info.json [hunk 11]
```diff
@@ -0,0 +1,60 @@
...
+    "processor": "atmega32u4",
+    "bootloader": "caterina",
```
Comment (waffle87) [inline keyboards/handwired/polly40/info.json:12]:
```suggestion
    "bootloader": "caterina",
    "diode_direction": "COL2ROW",
```

File: keyboards/handwired/polly40/readme.md [hunk 12]
```diff
@@ -0,0 +1,17 @@
...
+* Keyboard Maintainer: Paul Enrico N. Viola (https://github.com/PollyV1)
+* Hardware Supported: Pro micro Atmega32u4 controller
+* Hardware Availability: https://github.com/PollyV1/Polly40/
+
+
+Make example for this keyboard (after setting up your build environment):
+
+    make polly/default:default
+
+See the [build environment setup](https://docs.qmk.fm/#/getting_started_build_tools) and the [make instructions](https://docs.qmk.fm/#/getting_started_make_guide) for more information. Brand new to QMK? Start with our [Complete Newbs Guide](https://docs.qmk.fm/#/newbs).
```
Comment (waffle87) [inline keyboards/handwired/polly40/readme.md:17]: This file's contents need to conform to the [template](https://github.com/qmk/qmk_firmware/blob/master/data/templates/keyboard/readme.md?plain=1).

**Review by waffle87 — CHANGES_REQUESTED** [review]

File: keyboards/handwired/polly40/keymaps/default/keymap.c [hunk 13]
```diff
@@ -0,0 +1,49 @@
...
+        KC_1,    KC_2,    KC_3,    KC_4,    KC_5,    KC_6,    KC_7,    KC_8,    KC_9,    KC_0,    KC_ENT,  
+        KC_PSCR, KC_MPLY, KC_VOLD, KC_VOLU, KC_INS,  KC_DEL,  KC_COMM, KC_DOT,  KC_SLSH, KC_PGUP, MO(0),   
+        KC_LCTL, KC_LGUI, KC_LALT, KC_SPC,  KC_PGDN, KC_HOME  
+    ),
+
+    [2] = LAYOUT(
+        KC_ESC,  KC_1,    KC_2,    KC_3,    KC_R,    KC_T,    KC_Y,    KC_U,    KC_I,    KC_O,    KC_P,    KC_INS,  
+        KC_TAB,  KC_A,    KC_S,    KC_D,    KC_F,    KC_G,    KC_H,    KC_J,    KC_K,    KC_UP,   KC_DEL,  
+        KC_LSFT, KC_Z,    KC_X,    KC_C,    KC_V,    KC_B,    KC_N,    KC_LEFT, KC_DOWN, KC_RGHT, MO(0),   
+        KC_LCTL, KC_LGUI, KC_LALT, KC_SPC,  KC_HOME, _______
```
Comment (waffle87) [inline keyboards/handwired/polly40/keymaps/default/keymap.c:39]: Momentary layer toggles require a transparent key beneath them, and you're only able to go upwards in the layer stack, not downwards. eg. layer 0 cannot be accessed from layer 1, like you have here.
```suggestion
        KC_LCTL, KC_LGUI, KC_LALT, KC_SPC,  MO(2),   MO(1)    
    ),

    [1] = LAYOUT(
        KC_F1,   KC_F2,   KC_F3,   KC_F4,   KC_F5,   KC_F6,   KC_F7,   KC_F8,   KC_F9,   KC_F10,  KC_F11,  KC_F12,  
        KC_1,    KC_2,    KC_3,    KC_4,    KC_5,    KC_6,    KC_7,    KC_8,    KC_9,    KC_0,    KC_ENT,  
        KC_PSCR, KC_MPLY, KC_VOLD, KC_VOLU, KC_INS,  KC_DEL,  KC_COMM, KC_DOT,  KC_SLSH, KC_PGUP, _______,   
        KC_LCTL, KC_LGUI, KC_LALT, KC_SPC,  KC_PGDN, _______  
    ),

    [2] = LAYOUT(
        KC_ESC,  KC_1,    KC_2,    KC_3,    KC_R,    KC_T,    KC_Y,    KC_U,    KC_I,    KC_O,    KC_P,    KC_INS,  
        KC_TAB,  KC_A,    KC_S,    KC_D,    KC_F,    KC_G,    KC_H,    KC_J,    KC_K,    KC_UP,   KC_DEL,  
        KC_LSFT, KC_Z,    KC_X,    KC_C,    KC_V,    KC_B,    KC_N,    KC_LEFT, KC_DOWN, KC_RGHT, _______,   
        KC_LCTL, KC_LGUI, KC_LALT, KC_SPC,  _______, _______  
```

File: keyboards/handwired/polly40/keymaps/via/keymap.c [hunk 14]
```diff
@@ -0,0 +1,49 @@
...
+        KC_1,    KC_2,    KC_3,    KC_4,    KC_5,    KC_6,    KC_7,    KC_8,    KC_9,    KC_0,    KC_ENT,  
+        KC_PSCR, KC_MPLY, KC_VOLD, KC_VOLU, KC_INS,  KC_DEL,  KC_COMM, KC_DOT,  KC_SLSH, KC_PGUP, MO(0),   
+        KC_LCTL, KC_LGUI, KC_LALT, KC_SPC,  KC_PGDN, KC_HOME  
+    ),
+
+    [2] = LAYOUT(
+        KC_ESC,  KC_1,    KC_2,    KC_3,    KC_R,    KC_T,    KC_Y,    KC_U,    KC_I,    KC_O,    KC_P,    KC_INS,  
+        KC_TAB,  KC_A,    KC_S,    KC_D,    KC_F,    KC_G,    KC_H,    KC_J,    KC_K,    KC_UP,   KC_DEL,  
+        KC_LSFT, KC_Z,    KC_X,    KC_C,    KC_V,    KC_B,    KC_N,    KC_LEFT, KC_DOWN, KC_RGHT, MO(0),   
+        KC_LCTL, KC_LGUI, KC_LALT, KC_SPC,  KC_HOME, _______
```
Comment (waffle87) [inline keyboards/handwired/polly40/keymaps/via/keymap.c:39]:
```suggestion
        KC_LCTL, KC_LGUI, KC_LALT, KC_SPC,  MO(2),   MO(1)    
    ),

    [1] = LAYOUT(
        KC_F1,   KC_F2,   KC_F3,   KC_F4,   KC_F5,   KC_F6,   KC_F7,   KC_F8,   KC_F9,   KC_F10,  KC_F11,  KC_F12,  
        KC_1,    KC_2,    KC_3,    KC_4,    KC_5,    KC_6,    KC_7,    KC_8,    KC_9,    KC_0,    KC_ENT,  
        KC_PSCR, KC_MPLY, KC_VOLD, KC_VOLU, KC_INS,  KC_DEL,  KC_COMM, KC_DOT,  KC_SLSH, KC_PGUP, _______,   
        KC_LCTL, KC_LGUI, KC_LALT, KC_SPC,  KC_PGDN, _______  
    ),

    [2] = LAYOUT(
        KC_ESC,  KC_1,    KC_2,    KC_3,    KC_R,    KC_T,    KC_Y,    KC_U,    KC_I,    KC_O,    KC_P,    KC_INS,  
        KC_TAB,  KC_A,    KC_S,    KC_D,    KC_F,    KC_G,    KC_H,    KC_J,    KC_K,    KC_UP,   KC_DEL,  
        KC_LSFT, KC_Z,    KC_X,    KC_C,    KC_V,    KC_B,    KC_N,    KC_LEFT, KC_DOWN, KC_RGHT, _______,   
        KC_LCTL, KC_LGUI, KC_LALT, KC_SPC,  _______, _______  
```

File: keyboards/handwired/polly40/rules.mk [hunk 15]
```diff
@@ -0,0 +1,15 @@
...
+EXTRAKEY_ENABLE = yes       # Audio control and System control
+CONSOLE_ENABLE = yes        # Console for debug
+COMMAND_ENABLE = yes        # Commands for debug and configuration
+NKRO_ENABLE = yes           # Enable N-Key Rollover
+BACKLIGHT_ENABLE = no       # Enable keyboard backlight functionality
+RGBLIGHT_ENABLE = no        # Enable keyboard RGB underglow
+AUDIO_ENABLE = no           # Audio output
+
+# generated by KBFirmware JSON to QMK Parser
+# https://noroadsleft.github.io/kbf_qmk_converter/
```
Comment (waffle87) [inline keyboards/handwired/polly40/rules.mk:15]:
```suggestion
# This file intentionally left blank
```

**Review by waffle87 — CHANGES_REQUESTED** [review]

File: keyboards/handwired/polly40/.vscode/settings.json [hunk 16]
```diff
@@ -0,0 +1,3 @@
+{
```
Comment (waffle87) [inline keyboards/handwired/polly40/.vscode/settings.json:1]: This should not be included.

Comment (PollyV1) [reply keyboards/handwired/polly40/.vscode/settings.json:1]: I removed it already. it was from my ide

File: keyboards/handwired/polly40/readme.md [hunk 17]
```diff
@@ -0,0 +1,26 @@
+# Polly40
+
+![Polly40](https://i.imgur.com/QJCQcFAh.jpg)
+
+*A handwired 40% keyboard running a Pro micro Atmega32u4 controller*
+
+* Keyboard Maintainer: [Paul Enrico N. Viola](https://github.com/PollyV1)
+* Hardware Supported: *Atmega32u4 controller*
+* Hardware Availability: *https://shopee.ph/Pro-Micro-ATmega32U4-5V-16MHz-USB-Arduino-Compatible-i.20469516.225455828*
```
Comment (fauxpark) [inline keyboards/handwired/polly40/readme.md:9]:
```suggestion
A handwired 40% keyboard running a Pro micro Atmega32u4 controller

* Keyboard Maintainer: [Paul Enrico N. Viola](https://github.com/PollyV1)
* Hardware Supported: Atmega32u4 controller
* Hardware Availability: https://shopee.ph/Pro-Micro-ATmega32U4-5V-16MHz-USB-Arduino-Compatible-i.20469516.225455828
```

**Review by fauxpark — CHANGES_REQUESTED** [review]

**Comment by PollyV1** [comment]
Can I have a folder for my keyboards(Carpolly and polly40)? Carpolly is already in qmk master.

**Review by drashna — APPROVED** [review]

**Comment by waffle87** [comment]
> Can I have a folder for my keyboards(Carpolly and polly40)? Carpolly is already in qmk master.

Sure, but note keyboard moves are required to target the `develop` branch, which won't be merged into `master` for ~3 months.

**Comment by PollyV1** [comment]
> > Can I have a folder for my keyboards(Carpolly and polly40)? Carpolly is already in qmk master.
> 
> Sure, but note keyboard moves are required to target the `develop` branch, which won't be merged into `master` for ~3 months.

Oh ok, I'll leave it as it is for now

File: keyboards/handwired/polly40/polly40.h [hunk 18]
```diff
@@ -0,0 +1,36 @@
...
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#pragma once
+
+#include "quantum.h"
+
+#define XXX KC_NO
+
+#define LAYOUT( \
```
Comment (fauxpark) [inline keyboards/handwired/polly40/polly40.h:23]: If you set the matrix positions in the info.json layout, you can get rid of these two files.

Comment (PollyV1) [reply keyboards/handwired/polly40/polly40.h:23]: have not done it yet. i forgot. reverted the files first

**Comment by PollyV1** [comment]
Thanks guys!

**Pull request merged**